code/code/spec/spec_objs_thief_quest.cc

```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+
+#include "being.h"
+#include "comm.h"
+#include "extern.h"
+#include "obj_base_weapon.h"
+
+int thiefQuestWeapon(TBeing *victim, cmdTypeT command, const char *arg, TObj *object, TObj *) {
+  wearSlotT limb = WEAR_NOWHERE;
+
+  if (command != CMD_STAB && command != CMD_BACKSTAB && command != CMD_SLIT) return false;
+  if (command == CMD_STAB) limb = static_cast<wearSlotT>(reinterpret_cast<uintptr_t>(arg));
+  if (command != CMD_STAB && strcmp(arg, "-special-")) return false;
+  if (!victim || !object) return false;
+
+  limb = command == CMD_BACKSTAB ? WEAR_BACK
+              : command == CMD_SLIT   ? WEAR_NECK
+                                      : limb;
+
+  if (limb == WEAR_NOWHERE || limb == HOLD_RIGHT || limb == HOLD_LEFT || limb == MAX_WEAR)
+    return false;
+
+  auto limbDescription = victim->describeBodySlot(limb);
+
+  // 10% chance on stab, 50% on backstab/slit
+  if (command == CMD_STAB) {
+    if (::number(0, 9)) return false;
+  } else if (::number(0, 1))
+    return false;
+
+  auto weapon = dynamic_cast<TBaseWeapon *>(object);
+  auto thief = weapon ? dynamic_cast<TBeing *>(weapon->equippedBy) : nullptr;
+  if (!weapon || !thief) return false;
+
+  auto damage = 0;
```

Comment:
int is 1 less character to type :stuck_out_tongue:

Reply:
Yeah, this was before I learned that stuff about auto from you the other day. I'm gonna change them back to explicit types and just use auto from now on where it actually does save some typing/adds clarity.

Reply:
Resolved in https://github.com/sneezymud/sneezymud/pull/368/commits/7cd35cc3ce8966bdb70498453540f0af242e6778